nexus/reconfigurator/planning/src/blueprint_builder/clickhouse.rs

```diff
@@ -143,9 +137,27 @@ impl ClickhouseAllocator {
         // If we fail to retrieve any inventory for keepers in the current
         // collection than we must not modify our keeper config, as we don't
         // know whether a configuration is ongoing or not.
+        //
+        // There is an exception to this rule: on *new* clusters that have
+        // keeper zones deployed buty do not have any keepers running we must
```

Comment:
This actually has me thinking that we should gate removal of the last `ClickhouseKeeper` node by not allowing it to be expunged. However, that seems a lot easier said than done. I think in general this is not really a problem as we will have 5 of them and we won't expunge them all at once. Each time one is expunged a new one will be provisioned to reach the target in the same way we deal with CRDB.